lib/faker/phone/en.ex

```diff
@@ -0,0 +1,46 @@
+defmodule Faker.Phone.En do
+  @moduledoc """
+  This follows the rules outlined in the North American Numbering Plan
+  at https://en.wikipedia.org/wiki/North_American_Numbering_Plan.
+
+  The NANP number format may be summarized in the notation NPA-NXX-xxxx:
+
+  The allowed ranges for NPA (area code) are: [2–9] for the first digit, and
+  [0-9] for the second and third digits. The NANP is not assigning area codes
+  with 9 as the second digit.
+
+  The allowed ranges for NXX (central office/exchange) are: [2–9] for the first
+  digit, and [0–9] for both the second and third digits (however, in geographic
+  area codes the third digit of the exchange cannot be 1 if the second digit is
+  also 1).
+
+  The allowed ranges for xxxx (subscriber number) are [0–9] for each of the four
+  digits.
+  """
+
+  @doc """
+  Returns a random US phone number, optionally separated by `separator`
+  """
+  @spec us_phone(separator :: String.t) :: String.t
+  def us_phone(separator \\ "-") do
+    "#{area_code}#{separator}#{central_office}#{separator}#{subscriber_number}"
+  end
+
+  defp area_code do
+    [digit(2, 9), digit(0, 8), digit(0, 9)] |> Enum.join
+  end
+
+  defp central_office do
+    second_dig = digit(0, 9)
+    third_dig =
+      case second_dig do
+        1 -> digit(2, 9)
+        _ -> digit(1, 9)
+      end
+    [digit(2, 9), second_dig, third_dig] |> Enum.join
+  end
+
+  defp subscriber_number, do: Faker.format("####")
+
+  defp digit(min, max), do: :crypto.rand_uniform(min, max)
```

Comment:
this is good one, probably we want to put it in `Faker` top module to use it across lib.